psydac/linalg/solvers.py

```diff
@@ -946,7 +946,7 @@ def solve(self, b, out=None):
         rp.copy(out=rp0)
 
         # squared residual norm and squared tolerance
-        res_sqr = r.dot(r)
+        res_sqr = r.dot(r).real
```

Comment:
if `r = 1+1j`, then this will probably give `res_sqr = 0` (assuming that `dot` is a regular product). I don't think this will be correct. An inner product should be used instead.

Reply:
At the moment, the `dot` product of Psydac  does the complex conjugate when the underlying space field is complex. 
So in the case `r=v+1j*v`  (`v` is a `Vector` with only ones), then `res_sqr = 2*size(v) + 0j` .
In other to avoid the warning when taking `np.sqrt`, we can take the real part.

Reply:
Ok, so why not put the real conversion in the `dot` product (and call it `inner` for clarity) ?

Reply:
I think that this is quite a technical change, so in my opinion it would require a different PR. It would impact many places in the code and has to be consistent with Sympde. Maybe we can discuss it in the next Psydac meeting.